clean-code/src/main/java/code_smell/exchange_rates/ExchangeRatesProvider.java

```diff
@@ -0,0 +1,203 @@
+                    rate = String.valueOf((Double) ((HashMap) l.get(i)).get("mid"));
+                }
+            }
+        } else {
+            con = (HttpURLConnection) new URL("https://api.nbp.pl/api/exchangerates/tables/b/").openConnection();
+            con.setRequestMethod("GET");
+            //read response
+            bis = new BufferedInputStream(con.getInputStream());
+            buf = new ByteArrayOutputStream();
+            for (int result = bis.read(); result != -1; result = bis.read()) {
+                buf.write((byte) result);
+            }
+            response = buf.toString("UTF-8");
+            if (response.contains(cur)) {
+                List l = ((JSONArray) ((JSONObject) new JSONArray(response).get(0)).get("rates")).toList();
+                for (int i = 0; i < l.size(); i++) {
+                    if (((HashMap) l.get(i)).get("code").equals(cur)) {
+                        rate = String.valueOf((Double) ((HashMap) l.get(i)).get("mid"));
+                    }
+                }
+            } else {
+                con = (HttpURLConnection) new URL("https://api.nbp.pl/api/exchangerates/tables/a/").openConnection();
+                con.setRequestMethod("GET");
+                //read response
+                bis = new BufferedInputStream(con.getInputStream());
+                buf = new ByteArrayOutputStream();
+                for (int result = bis.read(); result != -1; result = bis.read()) {
+                    buf.write((byte) result);
+                }
+                response = buf.toString("UTF-8");
+                if (response.contains(cur)) {
+                    List l = ((JSONArray) (new JSONObject(response)).get("rates")).toList();
+                    if (((HashMap) l.get(0)).get("code").equals(cur)) {
+                        rate = String.valueOf((Double) ((HashMap) l.get(0)).get("mid"));
+                    } else {
+                        rate = "NOT_FOUND";
+                    }
+                }
+            }
+        }
+
+        // currency conversion is found here
+        //https://fastspring.com/blog/how-to-format-30-currencies-from-countries-all-over-the-world/
+        //currently we support only currencies EUR, USD, CAD, JPY
+        /*
+         *
+         * Abbreviation/Code: CAD
+         * Symbol: $
+         * Example: $ 1,234.56
+         *
+         * Abbreviation/Code: EUR
+         * Symbol: €
+         * Example: €1.234,56
+         *
+         * Abbreviation/Code: JPY
+         * Symbol: ¥
+         * Example: ¥ 1,234.56
+         *
+         * Abbreviation/Code: USD
+         * Symbol: $
+         * Example: $1,234.56
+         *
+         * Abbreviation/Code: MXN
+         * Symbol: $
+         * Example: $ 1,234.56
+         *
+         * Abbreviation/Code: SEK
+         * Symbol: kr
+         * Example: 1.234,56 kr
+         *
+         * */
+        Double amount = null;
+        if(!rate.equals("NOT_COUND")) {
```

Comment:
dobrze wynieść wszystkie Stringi określające tekst wiadomości do zmiennych finalnych (jak w komentarzu powyżej), 
dodatkowo popraw literówkę - zapewne miałes na mysli "NOT FOUND"